### ISSUE TITLE
Unable to compile (using solc 0.7.6) or migrate SimpleStorage to evm (ganache)

### ISSUE THREAD

**Issue by jensendarren** [truffle-box/optimism-box#17]
## Overview

I experienced two issues with this truffle box - with **compiling** and **migrating** as explained below:

### Compiling error with Solc 0.7.6

When compiling using solc 0.7.6 there is the following error:

```
> Compiling ./contracts/ethereum/SimpleStorage.sol
TypeError: Cannot convert undefined or null to object
```

**SOLUTION**: When changing the solc compiler to `0.7.3` (and possibly other versions) then the contract compiled successfully.

### Migration error due to missing migrations contract

When migrating the contracts then there is the following error:

```
Error: Could not find built Migrations contract: Could not find artifacts for Migrations from any sources
```

**SOLUTION**: When adding a `Migrations.sol` contract to the project then the migrations work. I notice a [Migrations contract is included in other Truffle boxes](https://github.com/truffle-box/react-box/blob/master/contracts/Migrations.sol) so I copied this across to my project.

Finally with that in place I am able to run the tests, using `truffle test` which pass:

````
  Contract: SimpleStorage
    ✓ ...should store the value 89. (156ms)


  1 passing (198ms)
````

Please confirm my above actions are required or if I've missed something! 

Many thanks.

**Comment by fainashalts** [comment]
Hi there @jensendarren! I will take a closer look at this on Monday, but some initial thoughts: 

The compilation error may be due to old null references from a previous migration -- if you try deleting your build directory and compiling again I wonder if the error persists? 

As for the Migrations contract, that should not be necessary in newer versions of Truffle. If you are getting this error and have the latest Truffle this may be a bug.

Can you share which Truffle version you are on? That will help me dig in to this further. Thanks so much!

**Comment by jensendarren** [comment]
Hi @fainashalts - thank you for your quick response! 

Thank for for sharing about the updated Truffle that does not require the Migrations contract. That's awesome! 

Meanwhile, this is the version that I've been using so far:

```
Truffle v5.1.14-nodeLTS.0 (core: 5.1.13)
Solidity - 0.7.3 (solc-js)
Node v14.16.0
Web3.js v1.2.1
```

I'll try updating Truffle and run through the steps again.

**Comment by jensendarren** [comment]
Hi @fainashalts 

I removed all the build artifacts, removed the `Migrations.sol` contract that I added my project, updated truffle to version `v5.4.10 (core: 5.4.10)`, reset to use solc version `0.7.6`, followed the steps and everything worked as expected!

So indeed the issue was my local version of Truffle. The version that I updated to is as follows:

```
Truffle v5.4.10 (core: 5.4.10)
Solidity - 0.7.6 (solc-js)
Node v14.16.0
Web3.js v1.5.2
```

Many thanks for your support!
